data-structures/ds-avltree/c/AVLRouteInC.c

```diff
@@ -0,0 +1,255 @@
+/*
+by Felipe Gabriel
+this code goes through the tree and prints the AVL tree in order, post order and pre order
+this code have all the main functions off a AVL tree 
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+
+typedef struct node{
+	int num;
+	struct node* left;
+	struct node* right;
+}Node;
+
+typedef Node* TreeBin;
+
+TreeBin * newTree(){
+	TreeBin* root = (TreeBin*) malloc(sizeof(TreeBin));
+	if(root)
+	    (*root) = NULL;
+
+	return root;
+}
+
+void clearNode(Node* node){
+	if(!node)	return;
+	clearNode(node->left);
+	clearNode(node->right);
+	free(node);
+	node = NULL;
```

Comment:
Codacy found an issue: [Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6412547)